geo/src/algorithm/convex_hull/mod.rs

```diff
@@ -77,7 +80,21 @@ where
     // are to be included.
     let mut ls: Vec<Coordinate<T>> = points.to_vec();
```

Comment:
Is it possible to avoid this extra `Vec` allocation?  Can you sort the input mut-slice directly?

Reply:
We can sort the mut-slice directly, however we can't do `ls.remove(1);` on a slice, so I don't see how we can do it without creating a new `Vec`.

@rmanoka I find it weird that the convex hull algorithms (`trivial_hull`, `graham_hull` and `quick_hull`) takes `points: &mut [Coordinate<T>]` as input.
We don't guarantee anything about how `points` has been mutated by the algorithm, right?

So wouldn't it be better to take either `&[Coordinate<T>]` or `Vec<Coordinate<T>>` as input?

Reply:
I think we need mutable points for the [`partition_slice`](https://github.com/georust/geo/blob/main/geo/src/utils.rs#L11-L32) function which reorders coordinates based on a predicate?

Reply:
> I think we need mutable points for the [`partition_slice`](https://github.com/georust/geo/blob/main/geo/src/utils.rs#L11-L32) function which reorders coordinates based on a predicate?

That would also work if we took input as `Vec<Coordinate<T>>`.

Reply:
> > I think we need mutable points for the [`partition_slice`](https://github.com/georust/geo/blob/main/geo/src/utils.rs#L11-L32) function which reorders coordinates based on a predicate?
> 
> That would also work if we took input as `Vec<Coordinate<T>>`.

I am confused – could you say more about what you have in mind here?

Reply:
> > > I think we need mutable points for the [`partition_slice`](https://github.com/georust/geo/blob/main/geo/src/utils.rs#L11-L32) function which reorders coordinates based on a predicate?
> > 
> > 
> > That would also work if we took input as `Vec<Coordinate<T>>`.
> 
> I am confused – could you say more about what you have in mind here?

So there are basically three options for taking slices as input in rust:
- Owned: `Box<[Coordinate<T>]>` or `Vec<Coordinate<T>>`. Makes sense if we need to consume the input to do the computation.
- Borrowed: `&[Coordinate<T>]`. Makes sense if we only need to read the input and consuming the input (as in "Owned") doesn't speed up the computation.
- Mutably borrowed: `&mut [Coordinate<T>]`. Makes sense if the computation needs to change the input, but also the user of the function should (at least in some situations) be interested in the mutated input. If mutated input is never useful for the user, we should prefer to use an owned input.

I'm arguing that the mutated `points` input is never useful for the user and thus to prevent the user from accidentally using the now garbage list of points and to make the api more clear, we should choose the "Owned" option.

Reply:
Yeah, the function signature was primarily whatever works for the implementation.  Note that the usage is anyway via the trait.  If it simplifies code without extra allocations, we can move to something more intuitive.

Reply:
> Yeah, the function signature was primarily whatever works for the implementation. Note that the usage is anyway via the trait. If it simplifies code without extra allocations, we can move to something more intuitive.

Makes sense. I don't think that change is relevant to this pull request.
Do I need to fix anything else before this can get merged?